x-pack/plugins/observability/public/data_handler.ts

```diff
@@ -19,25 +19,27 @@ interface FetchDataParams {
 export type FetchData<T extends FetchDataResponse = FetchDataResponse> = (
   fetchDataParams: FetchDataParams
 ) => Promise<T>;
+
 export type HasData = () => Promise<boolean>;
 
-interface DataHandler {
-  fetchData: FetchData;
+interface DataHandler<T extends ObservabilityApp = ObservabilityApp> {
+  fetchData: FetchData<ObservabilityFetchDataResponse[T]>;
   hasData: HasData;
 }
 
 const dataHandlers: Partial<Record<ObservabilityApp, DataHandler>> = {};
 
-export type RegisterDataHandler<T extends ObservabilityApp = ObservabilityApp> = (params: {
-  appName: T;
-  fetchData: FetchData<ObservabilityFetchDataResponse[T]>;
-  hasData: HasData;
-}) => void;
-
-export const registerDataHandler: RegisterDataHandler = ({ appName, fetchData, hasData }) => {
+export function registerDataHandler<T extends ObservabilityApp = ObservabilityApp>({
```

Comment:
Is `ObservabilityApp = ObservabilityApp` necessary?

```suggestion
export function registerDataHandler<T extends ObservabilityApp>({
```

Reply:
No, it's not. Thanks!